app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -43,25 +43,51 @@ window.GOVUK.analyticsGA4.analyticsModules = window.GOVUK.analyticsGA4.analytics
         return
       }
 
-      if (this.isMailToLink(href)) {
+      var linkAttributes = element.getAttribute('data-ga4-link')
+      if (linkAttributes) {
+        linkAttributes = JSON.parse(linkAttributes)
+        clickData = Object.assign(clickData, linkAttributes)
+
+        /* Since external links can't be determined in the template, we use _ as a signal
+        for our JavaScript to determine this value. */
+        if (clickData.external === '[populated-via-js]' && clickData.url) {
+          clickData.external = this.isExternalLink(clickData.url) ? 'true' : 'false'
+        }
+
+        if (clickData.link_method === '[populated-via-js]') {
+          clickData.link_method = this.getClickType(event)
+        }
+
+        if (clickData.index) {
+          clickData.index = parseInt(linkAttributes.index)
+        }
+        if (clickData.index_total) {
+          clickData.index_total = parseInt(linkAttributes.index_total)
+        }
+      } else if (this.isMailToLink(href)) {
         clickData.event_name = 'navigation'
         clickData.type = 'email'
         clickData.external = 'true'
+        clickData.url = href
+        clickData.text = element.textContent.trim()
+        clickData.link_method = this.getClickType(event)
       } else if (this.isDownloadLink(href)) {
         clickData.event_name = 'file_download'
         clickData.type = this.isPreviewLink(href) ? 'preview' : 'generic download'
         clickData.external = this.isExternalLink(href) ? 'true' : 'false'
+        clickData.url = href
+        clickData.text = element.textContent.trim()
+        clickData.link_method = this.getClickType(event)
```

Comment:
Non blocking comment - there's some repetition in this file with these lines. While they will never be executed more than once, we might want to think about a refactor at some point to make it a bit cleaner.

Reply:
@andysellick Good point - I think more link tracker types are going to be added soon (There's still attachment, internal , and navigation) so it might be safer to keep it like this for now, and optimise it once we know what those link types involve.